pkg/ingester/activeseries/active_series.go

```diff
@@ -100,13 +102,28 @@ func (c *ActiveSeries) purge(keepUntil time.Time) {
 	}
 }
 
-// Active returns the total number of active series, as well as a slice of active series matching each one of the
+func (c *ActiveSeries) ContainsRef(ref uint64) bool {
+	stripeID := ref % numStripes
+	return c.stripes[stripeID].containsRef(ref)
+}
+
+// Active returns the total number of active series. This method does not purge
+// expired entries, so ActiveWithMatchers should still be called periodically.
```

Comment:
What if we move the purge logic out of `ActiveWithMatchers()` too, into a dedicated function (which will be called after `ActiveWithMatcher()`? From a design perspective I think it's weird that `ActiveWithMatchers()` does the purge but `Active()` doesn't.

Reply:
Good idea, done in https://github.com/grafana/mimir/pull/5136/commits/8a71746a41eb6c428e1191a469b64e48aaefc367